eclair-core/src/main/scala/fr/acinq/eclair/db/DualDatabases.scala

```diff
@@ -0,0 +1,378 @@
+package fr.acinq.eclair.db
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder
+import fr.acinq.bitcoin.{ByteVector32, Crypto, Satoshi}
+import fr.acinq.eclair.channel._
+import fr.acinq.eclair.db.Databases.{FileBackup, PostgresDatabases, SqliteDatabases}
+import fr.acinq.eclair.db.DbEventHandler.ChannelEvent
+import fr.acinq.eclair.db.DualDatabases.runAsync
+import fr.acinq.eclair.db.pg._
+import fr.acinq.eclair.db.sqlite._
+import fr.acinq.eclair.io.Peer
+import fr.acinq.eclair.payment._
+import fr.acinq.eclair.router.Router
+import fr.acinq.eclair.wire.protocol.{ChannelAnnouncement, ChannelUpdate, NodeAddress, NodeAnnouncement}
+import fr.acinq.eclair.{CltvExpiry, MilliSatoshi, ShortChannelId}
+import grizzled.slf4j.Logging
+
+import java.io.File
+import java.util.UUID
+import java.util.concurrent.Executors
+import scala.collection.immutable.SortedMap
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+
+/**
+ * An implementation of [[Databases]] where there are two separate underlying db, one sqlite and one postgres.
+ * Sqlite is the main database, but we also replicate all calls to postgres.
+ * Calls to postgres are made asynchronously in a dedicated thread pool, so that it doesn't have any performance impact.
+ */
+case class DualDatabases(sqlite: SqliteDatabases, postgres: PostgresDatabases) extends Databases with FileBackup {
+
+  override val network: NetworkDb = DualNetworkDb(sqlite.network, postgres.network)
+
+  override val audit: AuditDb = DualAuditDb(sqlite.audit, postgres.audit)
+
+  override val channels: ChannelsDb = DualChannelsDb(sqlite.channels, postgres.channels)
+
+  override val peers: PeersDb = DualPeersDb(sqlite.peers, postgres.peers)
+
+  override val payments: PaymentsDb = DualPaymentsDb(sqlite.payments, postgres.payments)
+
+  override val pendingCommands: PendingCommandsDb = DualPendingCommandsDb(sqlite.pendingCommands, postgres.pendingCommands)
+
+  override def backup(backupFile: File): Unit = sqlite.backup(backupFile)
+}
+
+object DualDatabases extends Logging {
+
+  /** Run asynchronously and print errors */
+  def runAsync[T](f: => T)(implicit ec: ExecutionContext): Future[T] = Future {
+    Try(f) match {
+      case Success(res) => res
+      case Failure(t) =>
+        logger.error("postgres error:\n", t)
+        throw t
```

Comment:
Do we really want to throw here?
Since Sqlite is our primary DB, shouldn't we just swallow errors since this dual mode is only meant to let us monitor how postgres behaves in practice? This way we're sure it doesn't impact the running node, and can be deployed on mainnet safely (hopefully?).

Reply:
It already works that way, because this is all within a future that we are not waiting on. We have to provide a return type or an exception, but the `throw` is effectively ignored.